SurfaceTopography/Uniform/ScalarParameters.py

```diff
@@ -121,6 +137,47 @@ def rms_slope(topography, short_wavelength_cutoff=None, window=None,
             topography.dim))
 
 
+def rms_slope_from_profile(topography, short_wavelength_cutoff=None, window=None,
+                           direction=None):
+    """
+    Compute the root mean square amplitude of the height gradient of a
+    topography or line scan stored on a uniform grid.
+
+    Parameters
+    ----------
+    topography : :obj:`SurfaceTopography` or :obj:`UniformLineScan`
+        SurfaceTopography object containing height information.
+    short_wavelength_cutoff : float
+        All wavelengths below this cutoff will be set to zero amplitude.
+    window : str, optional
+        Window for eliminating edge effect. See scipy.signal.get_window.
+        Only used if short wavelength cutoff is set.
+        (Default: no window for periodic Topographies, "hann" window for
+        nonperiodic Topographies)
+    direction : str, optional
+        Direction in which the window is applied. Possible options are
+        'x', 'y' and 'radial'. If set to None, it chooses 'x' for line
+        scans and 'radial' for topographies. Only used if short wavelength
+        cutoff is set. (Default: None)
+
+    Returns
+    -------
+    rms_slope : float
+        Root mean square slope value.
+    """
+    if topography.is_domain_decomposed:
+        raise NotImplementedError(
+            "rms_slope not implemented for parallelized topographies")
+    if short_wavelength_cutoff is not None:
+        topography = topography.window(window=window, direction=direction)
+    if topography.dim == 1:
+        mask_function = None if short_wavelength_cutoff is None else \
+            lambda frequency: frequency[0] ** 2 < 1 / short_wavelength_cutoff ** 2
+        return np.sqrt((topography.derivative(1, mask_function=mask_function) ** 2).mean())
+    else:
+        raise ValueError('RMS slope can only be computed for line scans.')
```

Comment:
Why don't we do it linescan per linescan as for the profile ? 

I think it could be usefull in order to be able to compare the slopes of AFM data versus TEM, or because the AFM is artefacted in one direction.